brigade.js

```diff
@@ -0,0 +1,237 @@
+
+  goBuild.tasks.push(
+    "make build"
+  );
+
+  return goBuild;
+}
+
+function xbuild(e, p) {
+  var goBuild = new GoJob(`${projectName}-xbuild`);
+
+  goBuild.tasks.push(
+    "make xbuild-all"
+  );
+
+  return goBuild;
+}
+
+function test(e, p) {
+  var goTest = new GoJob(`${projectName}-test`);
+
+  goTest.tasks.push(
+    "make test-unit"
+  );
+
+  return goTest;
+}
+
+// TODO: we could refactor so that this job shares a mount with the build job above,
+// to remove the need of re-building before running test-cli
+function testIntegration(e, p) {
+  var goTest = new GoJob(`${projectName}-integrationtest`);
+  // Enable docker so that the daemon can be used for duffle commands invoked by test-cli
+  goTest.docker.enabled = true;
+
+  goTest.env.kubeconfig = {
+    secretKeyRef: {
+      name: "porter-kubeconfig",
+      key: "kubeconfig"
+    }
+  };
+
+  // Setup kubeconfig, fetch duffle, docker login, run tests
+  goTest.tasks.push(
+    "mkdir -p ${HOME}/.kube",
+    'echo "${kubeconfig}" > ${HOME}/.kube/config',
+    "apt-get update && apt-get install -y sudo",
+    "curl -fsSL https://raw.githubusercontent.com/fishworks/gofish/master/scripts/install.sh | bash",
+    "gofish init && gofish install duffle",
+    `docker login ${p.secrets.dockerhubRegistry} \
+      -u ${p.secrets.dockerhubUsername} \
+      -p ${p.secrets.dockerhubPassword}`,
+    `REGISTRY=${p.secrets.dockerhubOrg} make test-cli`
+  );
+
+  return goTest;
+}
+
+function publish(e, p) {
+  var goPublish = new GoJob(`${projectName}-publish`);
+
+  // TODO: we could/should refactor so that this job shares a mount with the xbuild job above,
+  // to remove the need of re-xbuilding before publishing
+
+  goPublish.env.AZURE_STORAGE_CONNECTION_STRING = p.secrets.azureStorageConnectionString;
+  goPublish.tasks.push(
+    "make xbuild-all publish"
+  )
+
+  return goPublish;
+}
+
+// Here we add GitHub Check Runs, which will run in parallel and report their results independently to GitHub
+function runSuite(e, p) {
+  checkRun(e, p, build, "Build").catch(e  => {console.error(e.toString())});
+  checkRun(e, p, xbuild, "Cross-Platform Build").catch(e  => {console.error(e.toString())});
+  checkRun(e, p, test, "Test").catch(e  => {console.error(e.toString())});
+  checkRun(e, p, testIntegration, "Integration Test").catch(e  => {console.error(e.toString())});
+}
+
+// **********************************************
+// Classes/Helpers
+// **********************************************
+
+// GoJob is a Job with Golang-related prerequisites set up
+class GoJob extends Job {
+  constructor (name) {
+    super(name);
+
+    const gopath = "/go";
+    const localPath = gopath + `/src/github.com/${projectOrg}/${projectName}`;
+
+    // Here using the large-but-useful deis/go-dev image as we have a need for deps
+    // already pre-installed in this image, e.g. helm, az, docker, etc.
+    // TODO: replace with lighter-weight image (Carolyn)
+    this.image = "deis/go-dev";
```

Comment:
@carolynvs did you find a better fit for a base image for our job pods here?

Reply:
Is it okay if we merge as-is and I'll make a slimmer image this week? I'd like to see this shiny new brigade job 🚢 without waiting for me to do hard work first.

Reply:
Definitely!  Let's do it.  We can improve iteratively.